contracts/utils/BaseLibEIP712.sol

```diff
@@ -10,16 +10,13 @@ abstract contract BaseLibEIP712 {
     string public constant EIP712_DOMAIN_VERSION = "v5";
 
     // EIP712Domain Separator
-    bytes32 public immutable EIP712_DOMAIN_SEPARATOR =
-        keccak256(
-            abi.encode(
-                keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
-                keccak256(bytes(EIP712_DOMAIN_NAME)),
-                keccak256(bytes(EIP712_DOMAIN_VERSION)),
-                getChainID(),
-                address(this)
-            )
-        );
+    bytes32 public immutable EIP712_DOMAIN_SEPARATOR;
```

Comment:
`EIP712_DOMAIN_SEPARATOR` probably should use the same fallback mechanism in `_getDomainSeparator`? In case dApps just read the variable `EIP712_DOMAIN_SEPARATOR`.

```solidity
bytes32 private immutable originalEIP712DomainSeparator;

constructor() {
        originalEIP712DomainSeparator = _buildDomainSeparator();
        originalChainId = getChainID();
}

function _getDomainSeparator() private view returns (bytes32) {
        if (getChainID() == originalChainId) {
            return originalEIP712DomainSeparator;
        } else {
            return _buildDomainSeparator();
        }
}

function EIP712_DOMAIN_SEPARATOR() external view returns (bytes32) {
        return _getDomainSeparator();
}
```

Reply:
Yeah it should although in most cases dapp should use `getEIP712Hash()` directly. Fixed in 246d8d1.